### ISSUE TITLE
dumb question about sam encoder

### ISSUE THREAD

**Issue by Jacky56** [Jumpat/SegAnyGAussians#2]
Hello gang,

I read the paper for your method but I have a dumb question.

Isn't it more computationally more efficient you directly embed the mask on the image then generate the model via 3dgs instead of applying the sam map during generation?

I'm probably misunderstanding something here, does your also reduce the number of ellipsoids on the point cloud compared to editing the images directly? I'm not clear on the benefits here.

**Comment by Jumpat** [comment]
Hi! Thanks for your interest! 
The key reason for training the 3D features for Gaussians is the ambiguity of the SAM-extracted masks. For an input image, SAM extracts many multi-granularity masks where exist a lot of overlap. And for different images for the same scene, the number of extracted masks is uncertain. 
Thus, simply training a mask field that embeds all masks is unfeasible.
Moreover, we use the SAM features to guide the learning of the 3D features. This is also because of the ambiguous 2D masks. Without the informative features and non-linear MLP, learning the multi-granularity segmentation is very hard. 
It is worth to mention that using SAM features is not time-consuming since before training we have pre-extracted the features of all images in the training set. During training, we only need to load them.

**Comment by Jacky56** [comment]
Thanks for your reply @Jumpat!
 
 I cannot wait for the code release to test it out!
